source/Octopus.Versioning/Octopus/OctopusVersionMaskParser.cs

```diff
@@ -21,27 +21,27 @@ public class OctopusVersionMaskParser
             // Get the major version number
             @$"(?<{Major}>\d+|{PatternIncrement}|{PatternCurrent})" +
             // Get the minor version number, delimited by a period, comma, dash or underscore
-            @$"(?:[.\-_](?<{Minor}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
+            @$"(?:\.(?<{Minor}>\d+|{PatternIncrement}|{PatternCurrent}))?" +
```

Comment:
These changes make the masks more strict, and align them with the output version generated by a mask, which has dots between version components and dash before the prerelease.